modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/actionItems/filterCreationActions.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ICreationActionItem} from '../../management_bar/components/CreationMenu';
+
+const filterCreationActions = ({creationActions, globalCollectionActions}: {
+	creationActions: Array<ICreationActionItem>,
+	globalCollectionActions: any
+}): Array<ICreationActionItem> | null => {
+	return creationActions
+		? creationActions.reduce(
+				(
+					creationActions: Array<ICreationActionItem>,
+					action: ICreationActionItem
+				) => {
+					if (action.data?.permissionKey) {
+						if (globalCollectionActions[action.data.permissionKey]) {
+							return [...creationActions, action];
+						}
+
+						return creationActions;
+					}
+
+					return [...creationActions, action];
+				},
+				[]
+		  )
+		: null;
```

Comment:
Can this `null` ever happen? `creationActions` is always an array, so it cannot be falsy

Reply:
I think that it could be. At least we cannot ensure that we will always have a creationMenu -> https://github.com/liferay-frontend/liferay-portal/pull/3781/files#diff-8bc70260aeeec91969d2ac7da9174557ebb931b6f352d71b4e4c066e2e6449abR412

Reply:
Hmm, I could have sworn there was a `if (creationActions) do filterCreationActions logic` in `FrontendDataSet.js`, in this PR  :thinking: 

That may of doing things may be cleaner. If `creationMenu` prop is unset, we can ignore any info from returned data, we know there will be no creation menu.

Reply:
Another point for my reasoning is TS in this PR:
```ts
const filterCreationActions = ({creationActions, globalCollectionActions}: {
	customActions: Array<ICreationActionItem>;
	globalCollectionActions: any
}): Array<ICreationActionItem> | null => {
```
It's not 
```
customActions: Array<ICreationActionItem> | undefined, // or null
```
TS does not complain, because the `null` check is an additional check. It only complains when a check is missing.

Reply:
Yes, makes sense using `Array<ICreationActionItem> | undefined;` for paramenter definition and returning value